pages/Article.jsx

```diff
@@ -1,14 +1,31 @@
 import { useParams } from 'react-router-dom'
 import { useEffect, useState } from 'react'
-import { getArticle, getComments } from '../api'
+import { getArticle, patchVotes } from '../api'
 import CommentList from '../components/CommentList'
 
 export default function Article () {
     const [currentArticle, setCurrentArticle] = useState(null)
     const [isError, setIsError] = useState(false)
     const [isLoading, setIsLoading] = useState(true)
+    const [isFailedRequest, setIsFailedRequest] = useState(false)
     const  { article } = useParams()
 
+    function handleVote (article, vote){
+        if (vote > 0) {
+            setCurrentArticle(prevArticle => {
+                return {...prevArticle, votes: prevArticle.votes+ 1}
```

Comment:
setting the whole article again feels a little overkill. How else might you divide your components to avoid having to change the current article state after upvoting?